### PR TITLE
PUD-1216: Add last_updated_by_user_id for recall in preparation for auditing

### PR DESCRIPTION
**Pull request by philsegal**
Deprecate with error `save(entity)` on recallRepository to ensure currentUserId is always updated on save.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt [hunk 1]
```diff
@@ -111,8 +112,8 @@ data class Recall(
   @Enumerated(STRING)
   val previousConvictionMainNameCategory: NameFormatCategory? = null,
   val hasDossierBeenChecked: Boolean? = null,
-  val previousConvictionMainName: String? = null,
   // MD: ideally this would be UserId, but hibernate/postgres does not make this easy :-(
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt:115]: Comment does not make sense after edit?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallService.kt [hunk 2]
```diff
@@ -83,25 +83,32 @@ class RecallService(
...
+      dossierTargetDate = calculateDossierTargetDate(updateRecallRequest.recallNotificationEmailSentDateTime)
+        ?: dossierTargetDate
     )
   }
 
   fun calculateDossierTargetDate(recallNotificationEmailSentDateTime: OffsetDateTime?): LocalDate? {
     return recallNotificationEmailSentDateTime?.let {
       var dossierTargetDate = it.toLocalDate().plusDays(1)
-      while (dossierTargetDate.dayOfWeek == DayOfWeek.SATURDAY || dossierTargetDate.dayOfWeek == DayOfWeek.SUNDAY || bankHolidayService.isHoliday(dossierTargetDate)) {
+      while (dossierTargetDate.dayOfWeek == DayOfWeek.SATURDAY || dossierTargetDate.dayOfWeek == DayOfWeek.SUNDAY || bankHolidayService.isHoliday(
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallService.kt:108]: Reformat confusing: convert clause to three lines each with one test.

File: src/main/resources/db/migration/V65__Add_last_updated_user_id.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+ALTER TABLE recall
+    ADD COLUMN last_updated_by_user_id UUID references user_details(id);
+
+update recall set last_updated_by_user_id = created_by_user_id
```
Comment (AndyNewton) [inline src/main/resources/db/migration/V65__Add_last_updated_user_id.sql:4]: Lowercase for SQL commands intentional?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/random/RandomUtilities.kt [hunk 4]
```diff
@@ -32,6 +32,7 @@ internal fun fullyPopulatedRecall(recallId: RecallId = ::RecallId.random(), crea
       assessedByUserId = (createdByUserId ?: ::UserId.random()).value,
       bookedByUserId = (createdByUserId ?: ::UserId.random()).value,
       createdByUserId = (createdByUserId ?: ::UserId.random()).value,
+      lastUpdatedByUserId = (createdByUserId ?: ::UserId.random()).value,
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/random/RandomUtilities.kt:35]: Tiny but feels like parameter `createdByUserId` could use a different name here: perhaps `recallUserId`?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt [hunk 5]
```diff
@@ -114,35 +114,38 @@ class RecallServiceTest {
     every { recallRepository.getByRecallId(recallId) } returns existingRecall
     val fixedClockTime = OffsetDateTime.now(fixedClock)
     val updatedRecallWithoutDocs = fullyPopulatedRecallWithoutDocuments.copy(lastUpdatedDateTime = fixedClockTime, recallNotificationEmailSentDateTime = fixedClockTime)
-    every { recallRepository.save(updatedRecallWithoutDocs) } returns updatedRecallWithoutDocs
+    val currentUserId = ::UserId.random()
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt:117]: Feels like a test setup `beforeEach()` would be a good step?

**Review by AndyNewton — DISMISSED** [review]
Approved with a few QQs.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt (hunk 1 above)
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt:115]: That's interesting. wonder how it got reordered!

**Review by AndyNewton — APPROVED** [review]

**Pull request merged**